lua/starfall/libs_cl/bass.lua

```diff
@@ -86,13 +141,42 @@ function bass_library.loadFile(path, flags, callback)
 				snd:Stop()
 				plyCount:free(instance.player, 1)
 			else
-				sounds[snd] = true
+				table.insert(soundList, snd)
+				sounds[snd] = { -- IGModAudioChannel is userdata, so we can't attach extra key-value pairs or functions to it directly.
```

Comment:
It would be good if we kept a separate table for 3D sounds so that line 94 can be removed.

Reply:
A static object above the instance block would make it a lot cleaner.
```lua
local soundCorrector = {
	sounds = {}
}

-- Adds sound and adds think hook if table was empty
function soundCorrector.add(snd)
end

-- Removes sound and removes think hook if table is empty
function soundCorrector.remove(snd)
end

-- Loops over sounds, calls soundCorrector.remove to remove them if invalid, calls the calc function for each valid one
function soundCorrector.think()
end

-- Do the math and volume correction if the volume is >1
function soundCorrector.calc(snd)
end
```